src/workerd/io/actor-cache-test.c++

```diff
@@ -1166,9 +1164,6 @@ KJ_TEST("ActorCache flush retry") {
   KJ_ASSERT(KJ_ASSERT_NONNULL(expectCached(test.get("corge"))) == "555");
   KJ_ASSERT(expectCached(test.get("grault")) == nullptr);
 
-  // Although the transaction didn't complete, the delete did, and so it resolves.
```

Comment:
If this promise definitely hasn't resolved in the new version of the code, can we encode that via a `poll()` like we've done below for `promise2`?

Frankly it looks worrying to me that `promise1` used to resolve and return 1 here -- we shouldn't have been trusting the results from a non-committed delete within a transaction, although perhaps we were cheating a bit and trusting it due to our knowledge that there's only ever one writer to a given actor's data at a time.